### ISSUE TITLE
About evaluation/rollouts json success value

### ISSUE THREAD

**Issue by YunOh21** [qiuboxiang/RoboCerebra#9]
Hello again :)

I finetuned openvla model for RoboCerebra recently, but the success bool values seem not that correct.

**[Case 1]**
The task is **place_down_the_cookies_into_basket**, but the robot just drops the cookies before even not that close to the basket.

The JSON file says:
```
{
  "video": "2025_11_10-03_29_18--robocerebra_Random_Disturbance--episode=0_step1--success=0--task=place_down_the_cookies_into_basket_in_contain_regi.mp4",
  "completed_subtasks": [
    "cookies_1"
  ],
  "success": true
}
```
However, the actual video is this:

https://github.com/user-attachments/assets/9e103063-aa72-4571-baa2-18cfbd8275e0

**[Case 2]**
The second example is the task **place_down_the_wine_onto_coffee_table**.

```
{
  "video": "2025_11_10-03_29_18--robocerebra_Random_Disturbance--episode=0_step10--success=1--task=place_down_the_wine_onto_coffee_table.mp4",
  "completed_subtasks": [
    "wine_bottle_1"
  ],
  "success": true
}
```
And the video is this:

https://github.com/user-attachments/assets/46b448df-e6ed-4093-a81f-b22ad4e02a0f

It seems that even one subtask is completed, then it says ```"success": true```

Can I change anything and see whether the real goal is completed?
Because I want to use the value as one of model evaluation metrics, it would be great if I can distinguish which is real success and partial success.

FYI, this evaluation was done with a model trained only one step, so frequent failures are expected, but seeing ```"success": true``` was unexpected.

I would appreciate any pointers.

**Comment by qiuboxiang** [comment]
Hi! :)

It looks like you’re running in resume + dynamic mode. This issue seems to have been caused by a counting logic problem when writing the JSON file — the logic didn’t properly ignore the resume state, which led to state mismatches and caused the kind of success flag mismatch you observed in Case 1.

This has been fixed in the latest code. The updated logic now automatically excludes resume frames from the counting process when writing JSON, which should resolve the mismatch issue seen in Case 1.

As for Case 2, it appears that under dynamic mode, an external disturbance caused the wine bottle to slip unexpectedly from the gripper and fall onto the table. Since the task goal was to place the wine bottle on the table, this kind of accidental success is currently still counted as a successful completion. Such outcomes are somewhat unavoidable due to the dynamic environment setup.

**Comment by YunOh21** [comment]
Thank you for the swift answer! 

I'll pull the latest code and then test again with a better trained model!

**Comment by YunOh21** [comment]
Hi, I got the latest code and the success values turns as expected for case 1 thanks to you.

However, I have a few concerns about training and evaluating with RoboCerebra.

I trained another model with 4000 steps, I assume whose meaning is 4000 epochs. It could be not enough amount, but with L1 loss converging, I assumed the model would show somewhat different behavior from the previous just 1 step trained model.

<img width="200" height="300" alt="L1 loss Image" src="https://github.com/user-attachments/assets/366cca54-8554-4c25-be58-c3501617c10a" />

But it seems like that my model doesn't show any difference and it acts very poorly in every case.

I saw the paper's metrics:

<img width="765" height="277" alt="Image" src="https://github.com/user-attachments/assets/74444534-ee24-4296-847f-6d23d65aa3aa" />

I think my model(OpenVLA finetuned with above TF data)'s performance is worse than any methods in this table, which seems to be zero. Below is the training command.

```
torchrun --standalone --nnodes 1 --nproc-per-node 1 vla-scripts/finetune.py \
  --vla_path "openvla/openvla-7b" \
  --data_root_dir "/home/ubuntu/tensorflow_datasets" \
  --dataset_name "robo_cerebra_dataset" \
  --run_root_dir "/home/ubuntu/ft-checkpoint" \
  --lora_rank 32 \
  --batch_size 8 \
  --grad_accumulation_steps 4 \
  --lr_warmup_steps 1000 \
  --num_steps_before_decay 2000 \
  --learning_rate 2e-4 \
  --image_aug True \
  --wandb_project "robocerebra" \
  --wandb_entity "yun5-personal" \
  --max_steps 10000 \
  --save_freq 1000 \
  --use_lora True \
2>&1 | tee -a train.log
```

The training took about 7 hours on L40S GPU (To be specific, AWS EC2 g6e.2xlarge).
I canceled during the training steps were getting 5k.

I uploaded this model in case I re-test or train from here: https://huggingface.co/Yun5/OpenVLA-RoboCerebra-L1-Proprio-4000/tree/main

So, my question is:

**1. Could I have converted the dataset somehow wrong?**
I downloaded the dataset and converted to RLDS referencing README.

```
# Install Hugging Face Hub if not already installed
pip install huggingface_hub

# Download the dataset (specify dataset type and enable resume)
huggingface-cli download qiukingballball/RoboCerebraBench --repo-type dataset --local-dir ./RoboCerebra_Bench --resume-download
```
```
cd rlds_dataset_builder/

# Step 1: Convert to HDF5
python regenerate_robocerebra_dataset.py \
  --robocerebra_raw_data_dir "/path/to/RoboCerebra_Bench/Ideal" \
  --robocerebra_target_dir "./converted_hdf5/robocerebra_ideal"

# Step 2: Convert to RLDS (disable CUDA to avoid initialization errors)
cd RoboCerebraDataset && CUDA_VISIBLE_DEVICES="" tfds build --overwrite
```

The converted data is here: https://huggingface.co/datasets/Yun5/RoboCerebra_TF

**2.** If the data were properly converted, then **the training steps and learning rate should be increased?**

<img width="736" height="82" alt="Image" src="https://github.com/user-attachments/assets/bd6cb7f3-ac3e-47c3-a124-5a8b267c7086" />

I see that your model was trained for 200K steps and the initial learning rate was 5e-5, so my 4000 steps must be not enough compared to that number.

**3. May I ask how did you set the training steps?**

I guess looking at loss graph is not okay for expecting training is enough.
I wonder **which metrics you checked**, or you tested the trained model with the benchmark many times so that the test result was the metric. Also, **what was the minimum training step number for a visible difference**?

If it's not too much, I would be grateful if you could provide the loss or training log you got from your experiments.
Or, any tip for training this dataset would be great.

Thank you as always.

**Comment by qiuboxiang** [comment]
For the dataset conversion, I recommend checking the mp4 files generated by regenerate_robocerebra_dataset.py to confirm that the videos look correct, the frames are in the right order, and they align properly with the actions. If the videos themselves are incorrect, the model will not learn anything meaningful.

Regarding training, we used an 8×A100 machine. For OpenVLA-OFT, the global batch size is calculated as batch_size × number_of_gpus. If you trained on a single GPU, your effective batch size becomes much smaller, while the learning rate remains the same as in our multi-GPU setup. This effectively increases the learning rate and often leads to poor or failed convergence. Based on your training log, the model indeed seems to have converged poorly.

During our training, we saved a checkpoint every 1000 steps and ran inference with those checkpoints to observe actual task performance, rather than relying solely on the loss curve. This was the primary way we evaluated whether the model was learning properly.Our loss typically converged to around 0.01.

**Comment by YunOh21** [comment]
I’ll try again with your suggestions in mind!

**Comment by YunOh21** [comment]
By the way, is there any plan to upload your trained model on huggingface?
I would like to see the paper's result, but training 200k steps on my side is extremely demanding for my current resources, so I would greatly appreciate it if a checkpoint could be shared.

**Comment by qiuboxiang** [comment]
Hi Yun :)
Currently, we don’t have plans to release the checkpoints, as they were trained on a small subset of scenes whose scene settings and object lists are very close to the evaluation environments.
The goal of this training was simply to obtain a usable low-level controller for closed-loop control, rather than to maximize its standalone performance.

**Comment by YunOh21** [comment]
Thanks a lot for taking the time to explain, especially this late at night.
From your paper, it looked like you also experimented with a planner or a hierarchical control framework.
I assume these components won’t be open-sourced either, right? It’s a bit of a shame — the work is really impressive.
Anyway, I'll do my best on my side :)

**Comment by qiuboxiang** [comment]
You’re right — at the moment, we don’t have plans to open-source the planner or hierarchical control components.
However, the appendix of our paper provides detailed prompts and methodological information, which you can follow to reproduce the main experimental results.
Truly appreciate your understanding and support!
